apiclient/blockchain.go

```diff
@@ -8,13 +8,89 @@ import (
 	"go.vocdoni.io/dvote/types"
 	"go.vocdoni.io/proto/build/go/models"
 	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
 )
 
 var (
 	//ErrTransactionDoesNotExist is returned when the transaction does not exist
 	ErrTransactionDoesNotExist = fmt.Errorf("transaction does not exist")
 )
 
+// TransactionsCost returns the current cost for all transactions
+func (c *HTTPclient) TransactionsCost() (*api.TransactionsCost, error) {
+	resp, code, err := c.Request(HTTPGET, nil, "chain", "transactions", "cost")
+	if err != nil {
+		return nil, err
+	}
+	if code != 200 {
+		return nil, fmt.Errorf("%s: %d (%s)", errCodeNot200, code, resp)
+	}
+	txscost := &api.TransactionsCost{}
+	err = json.Unmarshal(resp, txscost)
+	if err != nil {
+		return nil, err
+	}
+	return txscost, nil
+}
+
+// TransactionCost returns the current cost for the given transaction type
+func (c *HTTPclient) TransactionCost(txType models.TxType) (uint32, error) {
+	txscost, err := c.TransactionsCost()
+	if err != nil {
+		return 0, err
+	}
+	// TODO: lookup the field somehow dinamically instead of this giant switch case
+	switch txType {
+	case models.TxType_SET_PROCESS_STATUS:
+		return txscost.Costs.SetProcessStatus, nil
+	case models.TxType_SET_PROCESS_CENSUS:
+		return txscost.Costs.SetProcessCensus, nil
+	case models.TxType_SET_PROCESS_RESULTS:
+		return txscost.Costs.SetProcessResults, nil
+	case models.TxType_SET_PROCESS_QUESTION_INDEX:
+		return txscost.Costs.SetProcessQuestionIndex, nil
+	case models.TxType_SEND_TOKENS:
+		return txscost.Costs.SendTokens, nil
+	case models.TxType_SET_ACCOUNT_INFO_URI:
+		return txscost.Costs.SetAccountInfoURI, nil
+	case models.TxType_CREATE_ACCOUNT:
+		return txscost.Costs.CreateAccount, nil
+	case models.TxType_REGISTER_VOTER_KEY:
+		return txscost.Costs.RegisterKey, nil
+	case models.TxType_NEW_PROCESS:
+		return txscost.Costs.NewProcess, nil
+	case models.TxType_ADD_DELEGATE_FOR_ACCOUNT:
+		return txscost.Costs.AddDelegateForAccount, nil
+	case models.TxType_DEL_DELEGATE_FOR_ACCOUNT:
+		return txscost.Costs.DelDelegateForAccount, nil
+	case models.TxType_COLLECT_FAUCET:
+		return txscost.Costs.CollectFaucet, nil
+	default:
+		return 0, fmt.Errorf("transaction type unknown")
+	}
+}
+
+// SubmitTx POSTs the given tx to the API, and returns txHash and response
+func (c *HTTPclient) SubmitTx(stx *models.SignedTx) (types.HexBytes, []byte, error) {
```

Comment:
Is this method needed?

Usually in the apiclient we try to abstract the user from dealing with transactions. So if there is a need for sending a signed transaction, a specific method should be implemented. 

For dealing with multiple keys to perform different signatures you can use: `c.Clone("nexPrivKey").Whatever()`

Reply:
definitely not needed, actually unused. dropped it